tests/conftest.py

```diff
@@ -0,0 +1,215 @@
+from typing import Tuple
+
+import numpy as np
+import pytest
+
+from numpy.random import PCG64, Generator
+
+from gcm_filters.kernels import ALL_KERNELS, GridType
+
+
+# https://stackoverflow.com/questions/56374588/how-can-i-ensure-tests-with-a-marker-are-only-run-if-explicitly-asked-in-pytest
+# don't run overwrite_test_data marked test unless specifically asked to
+_require_explicit_marks = ["overwrite_test_data"]
+
+
+def pytest_collection_modifyitems(config, items):
+    keywordexpr = config.option.keyword
+    markexpr = config.option.markexpr
+    if keywordexpr or markexpr:
+        return  # let pytest handle this
+
+    for item in items:
+        for keyword in _require_explicit_marks:
+            if keyword in item.keywords:
+                skip = pytest.mark.skip(reason=f"{keyword} not explicitly selected")
+                item.add_marker(skip)
+
+
+_grid_kwargs = {
+    GridType.REGULAR: [],
+    GridType.REGULAR_AREA_WEIGHTED: ["area"],
+    GridType.REGULAR_WITH_LAND: ["wet_mask"],
+    GridType.REGULAR_WITH_LAND_AREA_WEIGHTED: ["wet_mask", "area"],
+    GridType.IRREGULAR_WITH_LAND: [
+        "wet_mask",
+        "dxw",
+        "dyw",
+        "dxs",
+        "dys",
+        "area",
+        "kappa_w",
+        "kappa_s",
+    ],
+    GridType.TRIPOLAR_REGULAR_WITH_LAND_AREA_WEIGHTED: ["wet_mask", "area"],
+    GridType.TRIPOLAR_POP_WITH_LAND: ["wet_mask", "dxe", "dye", "dxn", "dyn", "tarea"],
+}
+
+
+scalar_grids = [
+    GridType.REGULAR,
+    GridType.REGULAR_AREA_WEIGHTED,
+    GridType.REGULAR_WITH_LAND,
+    GridType.REGULAR_WITH_LAND_AREA_WEIGHTED,
+    GridType.IRREGULAR_WITH_LAND,
+    GridType.TRIPOLAR_REGULAR_WITH_LAND_AREA_WEIGHTED,
+    GridType.TRIPOLAR_POP_WITH_LAND,
+]
+tripolar_grids = [
+    GridType.TRIPOLAR_REGULAR_WITH_LAND_AREA_WEIGHTED,
+    GridType.TRIPOLAR_POP_WITH_LAND,
+]
+vector_grids = [GridType.VECTOR_C_GRID]
+
+
+def _make_random_data(shape: Tuple[int, int], seed: int) -> np.ndarray:
+    rng = Generator(PCG64(seed))
+    return rng.random(shape)
```

Comment:
Just noting a lesson learned: for the verification tests to work, the fixtures need to be deterministic. So here we are generating random data with a specific seed.

Reply:
Thanks for digging into this and getting to the bottom of the issue!